rust/trace/src/chrome_trace_dump.rs

```diff
@@ -82,6 +83,40 @@ where
     serde_json::from_reader(input).map_err(Error::Json)
 }
 
+/// Save tracing data to to supplied path, using the Trace Viewer format. Trace file can be opened
+/// using the Chrome browser by visiting the URL `about:tracing`. If `sorted_chronologically` is
+/// true then sort output traces chronologically by each trace's time of creation.
+pub fn save_traces<P: AsRef<Path>>(
+    output_path: P,
+    sorted_chronologically: bool,
+) -> Result<(), Error> {
+    let traces = if sorted_chronologically {
+        super::samples_cloned_sorted()
+    } else {
+        super::samples_cloned_unsorted()
+    };
+
+    match metadata(&output_path) {
+        Ok(md) => {
+            if md.is_file() {
+                return Err(Error::Io(IOError::from(IOErrorKind::AlreadyExists)));
+            }
+        }
+        Err(err) => {
+            return Err(Error::Io(err));
+        }
+    }
+
+    let mut trace_file = match File::create(&output_path) {
```

Comment:
We should be able to write all (most) of these as, `let thing = Failable::call()?;` and the errors will be correctly propagated up.